File: SWEA/D4/1210. ［S／W 문제해결 기본］ 2일차 － Ladder1/［S／W 문제해결 기본］ 2일차 － Ladder1.py
for _ in range(10):
	tc = int(input())
	grid =[list(map(int, input().split())) for _ in range(100)]
	x, y = 0, 0
	vis = [[0 for _ in range(100)] for _ in range(100)]
	for i in range(100):
		if grid[99][i]==2:
			x=99
			y=i
	vis[x][y] = 1
	while True:
		if x==0: break
		if 0<y<100 and grid[x][y-1]==1 and vis[x][y-1]==0:
			x, y = x, y-1
		elif 0<=y<99 and grid[x][y+1]==1 and vis[x][y+1]==0:
			x, y = x,y+1
		else:
			x,y = x-1,y
		vis[x][y] = 1
	
	print(f'#{tc} {y}')

File: SWEA/D4/1210. ［S／W 문제해결 기본］ 2일차 － Ladder1/［S／W 문제해결 기본］ 2일차 － Ladder1.java
import java.util.*;
import java.io.*;

public class Solution {
	static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
	static StringBuilder sb = new StringBuilder();
	static StringTokenizer st;
	
	static int[] dx = {0,0,-1};
	static int[] dy = {-1,1,0};

	public static void main(String[] args) throws Exception {
		int T = 10;
		while (T-- > 0) {
			int t = Integer.parseInt(input.readLine());
			int n = 100;
			int[][] grid = new int[n][n];
			boolean[][] vis = new boolean[n][n];

			for (int i = 0; i < n; i++) {
				st = new StringTokenizer(input.readLine());
				for (int j = 0; j < n; j++) {
					grid[i][j] = Integer.parseInt(st.nextToken());
				}
			}
			
			Deque<int[]> dq = new ArrayDeque<>();
			for (int i=0; i<n; i++) {
				if (grid[99][i] == 2) {
					dq.addLast(new int[] {99, i});
					vis[99][i] = true;
				}
			}
			
			int ans = 0;
			while (!dq.isEmpty()) {
				int[] now = dq.pollFirst();
				int x = now[0];
				int y = now[1];
				
				for (int d=0; d<3;d++) {
					int nx = x+ dx[d];
					int ny = y+dy[d];
					
					if (nx<0 || nx >= n || ny <0 || ny>=n) continue;
					if (vis[nx][ny]) continue;
					if (grid[nx][ny] == 0) continue;
					vis[nx][ny] = true;
					dq.addLast(new int[] {nx,ny});
					break;
				}
				
				if (x==0) ans = y;
			}
			
			sb.append("#").append(t).append(" ").append(ans).append("\n");
		}
		
		System.out.println(sb.toString());
	}
}